src/pages/groupComparison/ClinicalData.tsx

```diff
@@ -338,17 +350,40 @@ export default class ClinicalData extends React.Component<
             const axisData = this.clinicalDataPromise.result!;
             // for string like values include NA values if requested
             // TODO: implement for numbers
+            // Compare clinical data(axisData) samples with sample list, assign NA to difference
+            // If exclude overlapped patients and samples, remove overlapped samples from sample list
+            // If include overlapped patients and samples, keep all sample in sample list
+            let filteredSamples: Sample[] = [];
+            if (this.props.store.overlapStrategy === OverlapStrategy.EXCLUDE) {
+                const overlappingPatientsMap = getOverlappingPatientsMap(
+                    getOverlappingPatients(
+                        this.props.store._activeGroupsNotOverlapRemoved.result!
+                    )
+                );
+                const overlappingSamplesMap = getOverlappingSamplesMap(
+                    getOverlappingSamples(
+                        this.props.store._activeGroupsNotOverlapRemoved.result!
+                    )
+                );
+                filteredSamples = this.props.store.samples.result!.filter(
+                    sample =>
+                        !overlappingPatientsMap[sample.studyId]?.has(
+                            sample.patientId
+                        ) &&
```

Comment:
@alisman 
> actually, it's not totally clear to me why you have to put them into groups by study. they should have unique ID which should allow you to filter them without regard for study?

Yes it would be easier if it has unique id, but `overlappingPatients` (this.props.store._activeGroupsNotOverlapRemoved) doesn't have `uniquePatientKey`, it only has `studyId` and `patientId`. When we compare `overlappingPatients` with `this.props.store.samples`, we have to use both  `studyId` and `patientId`, so I put them in groups by study Id and put patient Ids in value. 
Same for `overlappingSamples`.

Reply:
I see.  Ok!

Reply:
would it be trivial to add the unique id wherever that thing is calculated?

Reply:
@leexgh i'm talking about this logic here.  like, move this seperate function in utils

Reply:
Done